cpp/src/arrow/flight/server.h

```diff
@@ -237,11 +244,13 @@ class ARROW_FLIGHT_EXPORT FlightServerBase {
   /// \brief Retrieve the schema for the indicated descriptor
   /// \param[in] context The call context.
   /// \param[in] request may be null
-  /// \param[out] schema the returned flight schema provider
-  /// \return Status
-  virtual Status GetSchema(const ServerCallContext& context,
-                           const FlightDescriptor& request,
-                           std::unique_ptr<SchemaResult>* schema);
+  /// \return Arrow result with the returned flight schema provider
+  virtual arrow::Result<std::unique_ptr<SchemaResult>> GetSchema(
```

Comment:
Why are we changing RPC handler signatures? IMO, let's leave these as-is for now. I'm not sure there's any real benefit in migrating them.

Reply:
@lidavidm Would you say the same about Client/ServerMiddlewareFactory, Client/ServerAuthHandler?